CastService.m

```diff
@@ -352,10 +352,14 @@ - (void)displayImage:(NSURL *)imageURL iconURL:(NSURL *)iconURL title:(NSString
         GCKImage *iconImage = [[GCKImage alloc] initWithURL:iconURL width:100 height:100];
         [metaData addImage:iconImage];
     }
-
+    
+    if(!self.castWebAppId){
+        self.castWebAppId = kGCKMediaDefaultReceiverApplicationID;
```

Comment:
You should not repeat this check all the time. It would be better to implement a custom getter to return the set value or the default value.